manufacturing-client/src/main.rs

```diff
@@ -296,10 +301,79 @@ async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType>
     }
     match mfg_string_type {
         MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+            log::debug!(
+                "msg_string_type value:{:#?} refers SerialNumber format ",
+                mfg_string_type
+            );
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            Ok(CborSimpleType::Text(serial_dmi))
+        }
+        MfgStringType::SerialNumberExt => {
+            log::debug!(
+                "msf_string_type value:{:#?} refers SerialNumberExt format ",
+                mfg_string_type
+            );
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+                .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
+                .map(|s| s.trim_end().to_owned())
+                .context("Error determining system serial number")?;
+            let output_ifaces = Command::new("ls")
+                .arg("/sys/class/net/")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .output()
+                .context("Failed to execute command")?;
+
+            let val = String::from_utf8_lossy(&output_ifaces.stdout).to_string();
+            let mut serial_mac = String::new();
+            for interface in val.lines() {
+                let path_mac_addr = format!("/sys/class/net/{interface}/address");
+                let mac = fs::read_to_string(path_mac_addr)
+                    .map(|s| s.trim_end().to_owned())
+                    .context("Error reading mac address")?;
+                log::info!(
+                    "mac read from file /sys/class/net/{interface}/address {:#?} ",
+                    mac
+                );
+                lazy_static! {
+                    static ref MAC_REGEX: Regex =
+                        Regex::new(r"^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$").unwrap();
+                }
+                if MAC_REGEX.is_match(&mac) {
+                    let mut non_zero_mac = false;
+                    for byte in mac.split(':') {
+                        let byte = u8::from_str_radix(byte, 16)?;
+                        if byte != 0 {
+                            non_zero_mac = true;
+                            break;
+                        }
+                    }
+                    if non_zero_mac {
+                        write!(serial_mac, "IFACE_{interface}={mac};").unwrap();
+                        log::debug!(
+                            "MAC address {:#?} of interface {:#?} \n",
```

Comment:
you can also use inline format args here if you fancy so

Reply:
there is no need for the \n

Reply:
added , thanks